vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt

```diff
@@ -310,12 +312,19 @@ class WidgetPostAPIHandler @AssistedInject constructor(@Assisted private val roo
 
         val params = HashMap<String, Any>()
         params["status"] = status
-        room.sendStateEvent(
-                eventType = EventType.PLUMBING,
-                stateKey = null,
-                body = params,
-                callback = createWidgetAPICallback(widgetPostAPIMediator, eventData)
-        )
+
+        GlobalScope.launch {
+            try {
+                room.sendStateEvent(
+                        eventType = EventType.PLUMBING,
+                        stateKey = null,
+                        body = params
+                )
+                widgetPostAPIMediator.sendSuccess(eventData)
```

Comment:
Should this be done inside the `try` block? If it throws, `sendError` will also be called, which isn't strictly the previous behaviour.
(Same for below)

Reply:
Thanks for pointing this out! Fixed in https://github.com/vector-im/element-android/pull/2500/commits/ed822becc68c16e03330b0139a7287861ccf72de